storages/backends/s3boto3.py

```diff
@@ -469,6 +469,7 @@ def _open(self, name, mode='rb'):
         return f
 
     def _save(self, name, content):
+        name, content = self.pre_save(name, content)
```

Comment:
As `pre_save` is the very first line of the method, I don't see the need for this at all. Couldn't a backend do the following?

```py
class MyS3Boto3Storage(S3Boto3Storage):
    def _save(self, name, content):
        name, content = self.do_my_special_thing()
        return super()._save(name, conent)
```

This is just as flexible, requires no special hooks and is compatible with recent releases.

Reply:
The entire idea of a hook in a library is to provide a way in which developers can run their own code in a guaranteed-to-be-stable way. This means that your proposal of directly subclassing the backends is not feasible, as there is no guarantee that the code in that method won't change (and break the subclass-patch).